test/e2e/scheduling/predicates.go

```diff
@@ -192,6 +194,135 @@ var _ = SIGDescribe("SchedulerPredicates [Serial]", func() {
 		verifyResult(cs, podsNeededForSaturation, 1, ns)
 	})
 
+	// This test verifies we don't allow scheduling of pods in a way that sum of limits +
+	// associated overhead is greater than machine's capacity.
+	// It assumes that cluster add-on pods stay stable and cannot be run in parallel
+	// with any other test that touches Nodes or Pods.
+	// Because of this we need to have precise control on what's running in the cluster.
+	// Test scenario:
+	// 1. Find the amount CPU resources on each node.
+	// 2. Create one pod with affinity to each node that uses 70% of the node CPU.
+	// 3. Wait for the pods to be scheduled.
+	// 4. Create another pod with no affinity to any node that needs 20% of the largest node CPU for the workload and
+	//    an overhead set as 30% of the largest node CPU.
+	// 5. Make sure this additional pod is not scheduled.
+
+	ginkgo.It("validates pod overhead is considered along with resource limits of pods that are allowed to run", func() {
+		WaitForStableCluster(cs, masterNodes)
+		nodeMaxAllocatable := int64(0)
+		nodeToAllocatableMap := make(map[string]int64)
+		for _, node := range nodeList.Items {
+			nodeReady := false
+			for _, condition := range node.Status.Conditions {
+				if condition.Type == v1.NodeReady && condition.Status == v1.ConditionTrue {
+					nodeReady = true
+					break
+				}
+			}
+			if !nodeReady {
+				continue
+			}
+			// Apply node label to each node
+			framework.AddOrUpdateLabelOnNode(cs, node.Name, "node", node.Name)
```

Comment:
i think the key "node" should be stored in a constant and use some sort of sha / randomization.
if someone writes a different test that also does something with a label key "node" either this or the other test will fail if the suite is running tests in parallel.

Reply:
AFAIU, most of the predicate tests shouldn't be run in parallel, though I'm not entirely clear on how this is enforced.  Hi @bsalamat  - can you help clarify?

Reply:
unless a test is tagged as `[Serial]` concurrent invocations of the suite will/can run this test in parallel. instead of tagging it `[Serial`] it's best to handle potential conflicts on the test side.

Reply:
This (and most of the ones for predicates) are tagged Serial. If node/pods are being modified at the same time in other tests, I am not sure if it will be possible to exercise specific aspects of predication/scheduler.